src/redux/books/books.js

```diff
@@ -1,24 +1,49 @@
+import * as apiCalls from '../../api/api';
+
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
-
+const FETCH_BOOKS = 'bookStore/books/FETCH_BOOKS';
 const initialState = [];
 
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+export const addBook = (payload) => async (dispatch) => {
+  await apiCalls.addBookToApi(payload);
+  dispatch({
+    type: ADD_BOOK,
+    payload,
+  });
+};
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+export const removeBook = (id) => async (dispatch) => {
+  await apiCalls.removeBookFromApi(id);
+  dispatch({
+    type: REMOVE_BOOK,
+    payload: id,
+  });
+};
+// eslint-disable-next-line
+export const fetchAllBooks = () => async (dispatch) => {
+  const apiData = await apiCalls.getAllBooksFromApi();
+  const payload = Object.entries(apiData).map(([key, value]) => {
+    const [books] = value;
+    return {
+      item_id: key,
+      ...books,
+    };
+  });
+  dispatch({
+    type: FETCH_BOOKS,
+    payload,
+  });
+};
 // eslint-disable-next-line
 const booksReducer = (state = initialState, action) => {
   switch (action.type) {
```

Comment:
- [ ] Don't disable eslint errors, please fix them without disabling it.

- [ ] Please fix the warnings that have been raised after `npm start`

![setup-7](https://user-images.githubusercontent.com/55635977/145472059-14bc29be-548e-4f79-8317-6b66edb5928d.png)